### PR TITLE
[FLINK-6521] Add per job cleanup methods to HighAvailabilityServices

### PR DESCRIPTION
**Pull request by zjureel**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServices.java [hunk 1]
```diff
@@ -113,6 +116,11 @@ public ZooKeeperHaServices(
 		this.runningJobsRegistry = new ZooKeeperRunningJobsRegistry(client, configuration);
 
 		this.blobStoreService = checkNotNull(blobStoreService);
+		try {
+			this.submittedJobGraphStore = ZooKeeperUtils.createSubmittedJobGraphs(client, configuration, executor);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/zookeeper/ZooKeeperHaServices.java]: We should not throw `RuntimeException` but instead a meaningful checked exception.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/HighAvailabilityServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.junit.Assert;
+import org.junit.rules.ExpectedException;
+
+public class HighAvailabilityServiceTest {
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/HighAvailabilityServiceTest.java]: If we let the test case extend from `TestLogger`, then we get nice testing log statement on Travis.

File: flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/HighAvailabilityServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+						new JobGraph("testSubmittedJob2"),
+						new JobInfo(ActorRef.noSender(), ListeningBehaviour.DETACHED, 0, Integer.MAX_VALUE));
+		submittedJobGraphStore.putJobGraph(jobGraph1);
+		submittedJobGraphStore.putJobGraph(jobGraph2);
+
+		zkHaService.cleanupData(jobGraph1.getJobId());
+
+		SubmittedJobGraph recoverJobGraph2 = submittedJobGraphStore.recoverJobGraph(jobGraph2.getJobId());
+		Assert.assertEquals(recoverJobGraph2.getJobId(), jobGraph2.getJobId());
+		thrown.expectMessage("Could not retrieve the submitted job graph state handle for /" +
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/HighAvailabilityServiceTest.java]: Could we rather check for the exception type? Matching exception messages is really brittle.

**Comment by tillrohrmann** [comment]
The test case `JobManagerHACheckpointRecoveryITCase.testCheckpointedStreamingProgramIncrementalRocksDB` seems to be failing on Travis. It might be something caused by the changes.

**Comment by zjureel** [comment]
I found the following kinda stuff from CI, and it seems not relevant to this issue, what do you think? @tillrohrmann 
```
Running org.apache.flink.test.recovery.JobManagerHACheckpointRecoveryITCase
java.lang.RuntimeException: org.apache.zookeeper.server.ZooKeeperServer class is frozen
	at javassist.CtClassType.checkModify(CtClassType.java:288)
	at javassist.CtBehavior.setBody(CtBehavior.java:432)
	at javassist.CtBehavior.setBody(CtBehavior.java:412)
	at org.apache.curator.test.ByteCodeRewrite.fixMethods(ByteCodeRewrite.java:91)
	at org.apache.curator.test.ByteCodeRewrite.<clinit>(ByteCodeRewrite.java:50)
	at org.apache.curator.test.TestingServer.<clinit>(TestingServer.java:33)
	at org.apache.flink.test.recovery.JobManagerHACheckpointRecoveryITCase.testCheckpointedStreamingProgram(JobManagerHACheckpointRecoveryITCase.java:350)
	at org.apache.flink.test.recovery.JobManagerHACheckpointRecoveryITCase.testCheckpointedStreamingProgramIncrementalRocksDB(JobManagerHACheckpointRecoveryITCase.java:336)
```

**Comment by tillrohrmann** [comment]
Closed for inactivity.

**Pull request closed without merging**